File: iglview/src/main/java/com/aaronlee/iglview/EGLContextFactory.java
package com.aaronlee.iglview;

import javax.microedition.khronos.egl.EGL10;
import javax.microedition.khronos.egl.EGLConfig;
import javax.microedition.khronos.egl.EGLContext;
import javax.microedition.khronos.egl.EGLDisplay;

/**
 * An interface for customizing the eglCreateContext and eglDestroyContext calls.
 * <p>
 * This interface must be implemented by clients wishing to call
 * {@link GLSurfaceView#setEGLContextFactory(EGLContextFactory)}
 */
public interface EGLContextFactory {
    EGLContext createContext(EGL10 egl, EGLDisplay display, EGLConfig eglConfig, int eglContextClientVersion);
    void destroyContext(EGL10 egl, EGLDisplay display, EGLContext context);
}

File: iglview/src/main/java/com/aaronlee/iglview/EGLContextFactory.kt
package com.aaronlee.iglview

import android.opengl.EGLConfig
import android.opengl.EGLContext
import android.opengl.EGLDisplay

/**
 * An interface for customizing the eglCreateContext and eglDestroyContext calls.
 *
 *
 * This interface must be implemented by clients wishing to call
 * [GLSurfaceView.setEGLContextFactory]
 */
interface EGLContextFactory {
    fun createContext(
        display: EGLDisplay?,
        eglConfig: EGLConfig?,
        eglContextClientVersion: Int
    ): EGLContext?

    fun destroyContext(display: EGLDisplay?, context: EGLContext?)
}